kernel/src/expressions/scalars.rs

```diff
@@ -229,4 +279,55 @@ mod tests {
         let s = Scalar::Decimal(123, 9, -3);
         assert_eq!(s.to_string(), "123000");
     }
+
+    fn assert_decimal(
+        raw: &str,
+        expect_int: i128,
+        expect_prec: u8,
+        expect_scale: i8,
+    ) -> Result<(), Box<dyn std::error::Error>> {
+        let s = PrimitiveType::Decimal(expect_prec, expect_scale);
+        match s.parse_scalar(raw)? {
+            Scalar::Decimal(int, prec, scale) => {
+                assert_eq!(int, expect_int);
+                assert_eq!(prec, expect_prec);
+                assert_eq!(scale, expect_scale);
+            }
+            _ => assert!(false, "Didn't parse as decimal"),
+        };
+        Ok(())
+    }
+
+    #[test]
+    fn test_parse_decimal() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
these include all the examples from https://docs.oracle.com/javase/8/docs/api/java/math/BigDecimal.html#BigDecimal-java.lang.String-

Open to other things to test